### ISSUE TITLE
Target Deployments from All Branches

### ISSUE THREAD

**Issue by simonireilly** [simonireilly/observatory-github-action#34]
In order to run the CI tool, we should provide a very good and intuitive readme.

- [x] Provide an example of running the CI action against a deployment that is triggered by a merge request
- [x] Provide an example of running the CI action against a pull request
- [x] Provide an example of using GitHub to compute the preview URL's for common tools:
  - [x] Vercel
  - [x] Netlify
  - [x] Custom configuration of the URL with parameters from a commit, or deployment

### Background

There are some useful tools that show how you can access the GitHub refs for deployment here: https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#deployment

Further, there is an API document that is in use for sending deployment information to GitHub: https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-deployment-status

It would be ideal if the deployment would send the URL, and it was available in the GitHub action, having been triggered by the deployment e.g.

```YAML
on:
  deployment:


jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Mozilla Observatory Github Action
        id: observatory
        uses: simonireilly/observatory-github-action@v0.0.0.canary-3
        with:
          web_host: ${{github.deployment.environment_url}}
      - name: Create commit comment
        uses: peter-evans/commit-comment@v1
        with:
          body: ${{ steps.observatory.outputs.observatory-report }}
```

**Comment by simonireilly** [comment]
This is possible by accessing the event on the GitHub context. It will contain the parameters of the event which has triggered the workflow:

https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context

**Comment by simonireilly** [comment]
closed by #73

**Comment by simonireilly** [comment]
Closed by #73
